tests/Unit/Options/Test_Options.cpp

```diff
@@ -147,11 +150,71 @@ SPECTRE_TEST_CASE("Unit.Options.NamedSimple.invalid", "[Unit][Options]") {
   opts.get<NamedSimple>();
 }
 
+namespace {
+struct Group {
+  static constexpr OptionString help = {"Group halp"};
+};
+
+struct GroupedTag {
+  using type = int;
+  static constexpr OptionString help = {"Tag halp"};
+  using group = Group;
+};
+
+struct OuterGroup {
+  static constexpr OptionString help = {"Outer group halp"};
+};
+
+struct InnerGroup {
+  static constexpr OptionString help = {"Inner group halp"};
+  using group = OuterGroup;
+};
+
+struct InnerGroupedTag {
+  using type = int;
+  static constexpr OptionString help = {"Inner tag halp"};
+  using group = InnerGroup;
+};
+
+struct OuterGroupedTag {
+  using type = int;
+  static constexpr OptionString help = {"Outer tag halp"};
+  using group = OuterGroup;
+};
+
+void test_options_grouped() {
+  {
+    INFO("Option groups");
+    Options<tmpl::list<GroupedTag, Simple>> opts("Overall help text");
+    opts.parse(
+        "Group:\n"
+        "  GroupedTag: 3\n"
+        "Simple: 2");
+    CHECK(opts.get<GroupedTag>() == 3);
+    CHECK(opts.get<Simple>() == 2);
+  }
+  {
+    INFO("Nested option groups");
+    Options<tmpl::list<InnerGroupedTag, OuterGroupedTag, Simple>> opts(
+        "Overall help text");
+    opts.parse(
+        "OuterGroup:\n"
+        "  InnerGroup:\n"
+        "    InnerGroupedTag: 3\n"
+        "  OuterGroupedTag: 1\n"
+        "Simple: 2");
```

Comment:
Good catch. I fixed the output to what I intended it to be, i.e. display all options tags and denoting the end of a group by a blank line. I think it would be good to actually list all tags in the help string, otherwise the help string becomes way less useful and one has to try-and-error one's way through the groups. However I agree the current format is not ideal, but can't think of a better way of separating groups right now (indenting would be nice, but then we can't really impose the line length limits anymore).

Reply:
I don't think people are going to understand this output.  Reading it involves things like counting the number of blank lines separating the options, even after one has guessed what that means.  (And I am not sure that I can reliably distinguish between different numbers of blank lines when that number is greater than one.)

I agree that having to dig down through the groups to get help is annoying, but it is consistent with how all the other options work.

Reply:
I agree with @wthrowe. I had a hard time figuring out what the error message was trying to tell me, and that's with very hintful names like "Outer" and "Inner"

Reply:
Yes as I said, it’s not ideal. I‘ll change it to only show top-level options.

Here‘s an idea (for a future PR): How about we add functionality that generates an input file for the executable, including help-string comments, that the user can then fill in?